### DISCUSSION TITLE
How to enable the C++ standrad library

### DISCUSSION THREAD

**Discussion by r0n9Y** [zephyrproject-rtos/zephyr#52964] [Q&A]
Based on the doc https://docs.zephyrproject.org/latest/develop/languages/cpp/index.html to use the full STL, It seems I need to rebuild the Zephyr SDK. But there is no instruction. Does anyone know where can I have the instruction? I also have an impression that there is a config that is able to enable a subset of C++ STL functionality, which sort of only has std::vector included. But I can not find the config name in the Guiconfig. Any suggestions

**Comment by damiandrzewicz** [comment]
Try following configuration:

`CONFIG_CPLUSPLUS=y
CONFIG_EXCEPTIONS=y
CONFIG_LIB_CPLUSPLUS=y
CONFIG_STD_CPP17=y
CONFIG_NEWLIB_LIBC=y`
